pkg/webhooks/webhookpatch.go

```diff
@@ -81,7 +81,7 @@ const delayedRetryTime = time.Second
 // - pass the existing k8s client
 // - use the K8S root instead of citadel root CA
 // - removed the watcher - the k8s CA is already mounted at startup, no more delay waiting for it
-func PatchCertLoop(injectionWebhookConfigName, webhookName, caBundlePath string, client kubernetes.Interface, stopCh <-chan struct{}) {
+func PatchCertLoop(injectionWebhookConfigName, webhookName, watchRev, caBundlePath string, client kubernetes.Interface, stopCh <-chan struct{}) {
```

Comment:
I think `injectionWebhookConfigName` means that even though we are getting updates triggered for any webhook with istio.io/rev, it will always patch <injectionWebhookConfigName>, not the one that was updated

Reply:
Ahh, so injectionWebhookConfigName should become the rev?

Reply:
I think current behavior is:
1) Patch injectionWebhookConfigName
2) Watch for changes to injectionWebhookConfigName
3) On update of injectionWebhookConfigName, patch it again

New behavior would be
1) Watch for changes to any webhook with istio.io/rev=<rev>
2) On update of a webhook, patch it

I think the initial patching is probably not actually needed. Likely there as an optimization which may or may not apply with the new logic